internal/brain/events.go

```diff
@@ -38,6 +38,25 @@ func CheckValidMove(position int16, tab [board.TOT_SIZE]int16, playerId int16) b
 	return false
 }
 
+// Only called from server; improves UI
+func getInvalidMovesForCurrentPlayer(tab *[board.TOT_SIZE]int16, playerId int16) []int16 {
+	var invalidMoves []int16
+	for idx := range *tab {
+		isValid := CheckValidMove(int16(idx), *tab, playerId)
+		if !isValid {
+			invalidMoves = append(invalidMoves, int16(idx))
+		}
+	}
+	return invalidMoves
+}
+
+func GetInvalidMovesForCurrentPlayer(playerId int16) {
+	Game.InvalidMovesForCurrentPlayer = nil
+	if Game.CurrentPlayer.AiStatus > 0 {
```

Comment:
Why is this supposed to return invalid moves only when player is an AI? Isn't it supposed to return that every time?

Reply:
The only purpose is to have an easy way to disable illegal moves on the front without making a separate request. Since the AI player will never try to make an illegal move, it's not relevant to him.

I was gonna just marshall it as another key/value along with global Game state returned from server, but since that looked annoying, I did it the lazy way and just added a prop to global Game.

Alternatives is I remove it from the global Game state (makes the most sense) or I can rename it (InvalidMovesForHumanFrontPlayer... meh)

What say you?

Reply:
It is ok then, no need to refoactor, but just make it AiSatus == 0 ?